qemu/tests/seabios_iommu.py

```diff
@@ -0,0 +1,121 @@
+                    test.error("Failed to get drive path of '%s'" % img)
+                did = drive_path[5:]
+                for mp in re.finditer(r'/dev/%s\d+\s+(\S+)\s+' % did, mounts):
+                    mount_points.append(mp.group(1))
+                else:
+                    mp = utils_disk.configure_empty_linux_disk(session,
+                                                               did, size)
+                    mount_points.extend(mp)
+        elif os_type == "windows":
+            size = params["data_image_size"]
+            disks = utils_disk.get_windows_disks_index(session, size)
+            disk_num = len(disks)
+            if disk_num < image_num:
+                err_msg = "set disk num: %d" % image_num
+                err_msg += ", get in guest: %d" % disk_num
+                test.fail("Fail to list all the volumes, %s" % err_msg)
+            if not utils_disk.update_windows_disk_attributes(session, disks):
+                test.fail("Failed to update windows disk attributes.")
+            for disk in disks:
+                drive_letter = utils_disk.configure_empty_windows_disk(session,
+                                                                       disk,
+                                                                       size)
+                if not drive_letter:
+                    test.fail("Fail to format disks.")
+                mount_points.extend(drive_letter)
+        else:
+            test.cancel("Unsupported OS type '%s'" % os_type)
+        return mount_points
+
+    image_list = params.objects("images")[1:]
+    image_num = len(image_list)
+
+    error_context.context("Check data disks in monitor!", logging.info)
+    monitor_info_block = vm.monitor.info_block(False)
+    blocks = ','.join(monitor_info_block.keys())
+    for image in image_list:
+        if image not in blocks:
+            test.fail("drive_%s is missing: %s!" % (image, blocks))
+
+    error_context.context("Read and write data on data disks!", logging.info)
+    iozone_test = generate_instance(params, session, 'iozone')
+    iozone_cmd = params["iozone_cmd"]
+    iozone_timeout = float(params.get("iozone_timeout", 1800))
+    try:
+        for mp in _get_mount_points():
+            iozone_test.run(iozone_cmd % mp, iozone_timeout)
+    finally:
+        iozone_test.clean()
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    [seabios] seabios support IOMMU for virtio-blk
+    [seabios] seabios support IOMMU for virtio-scsi
+    [seabios] seabios support IOMMU for virtio-net
+
+    this case will:
+    1) Boot guest with virtio devices and iommu is on.
+    2) Check 'info block'.
+    3) Read and write data on data disks.
+    4) Ping guest.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    if utils_misc.get_cpu_vendor(verbose=False) != 'GenuineIntel':
```

Comment:
get_cpu_vendor move to virttest.cpu. please use latest avocado codes. e.g. cpu.get_cpu_vendor()